crates/node/src/command.rs

```diff
@@ -55,71 +55,97 @@ impl SubstrateCli for Cli {
-                )?;
-            }
-        }
-    }
+    cli.run.run_cmd.force_authoring = true;
+    cli.run.run_cmd.alice = true;
+
+    // we can't set `--rpc-cors=all`, so it needs to be set manually if we want to connect with external
+    // hosts
+    cli.run.run_cmd.rpc_external = true;
+    cli.run.run_cmd.rpc_methods = RpcMethods::Unsafe;
+}
+
+fn set_testnet(cli: &mut Cli, testnet: Testnet) -> Result<(), String> {
+    // checks if it should retrieve and enable a specific chain-spec
+    let madara_path = cli.run.madara_path.clone().expect("Failed retrieving madara_path").to_str().unwrap().to_string();
+    let local_path = utils::get_project_path();
 
-    // TODO confirm with the CI that we are fetching all and fetch dynamically
-    // Issue #982
-    for file in constants::CAIRO_CONTRACTS_FILES {
-        let src_path = utils::get_project_path();
-        if let Ok(src_path) = src_path {
-            let src_path = src_path + "/configs/cairo-contracts/" + file;
-            utils::copy_from_filesystem(src_path, madara_path.clone() + "/cairo-contracts")?;
+    if testnet == Testnet::Sharingan {
+        if let Ok(ref src_path) = local_path {
+            let src_path = src_path.clone() + "/configs/chain-specs/testnet-sharingan-raw.json";
+            utils::copy_from_filesystem(src_path, madara_path.clone() + "/chain-specs")?;
+            cli.run.run_cmd.shared_params.chain = Some(madara_path + "/chain-specs/testnet-sharingan-raw.json");
         } else {
             utils::fetch_from_url(
-                constants::CAIRO_CONTRACTS_URL.to_string() + file,
-                madara_path.clone() + "/cairo-contracts",
+                constants::SHARINGAN_CHAIN_SPEC_URL.to_string(),
+                madara_path.clone() + "/configs/chain-specs/",
             )?;
+            cli.run.run_cmd.shared_params.chain = Some(madara_path + "/chain-specs/testnet-sharingan-raw.json");
         }
     }
 
-    if let (Some(chain_spec_url), None) = (cli.run.chain_spec_url.clone(), cli.run.testnet) {
-        utils::fetch_from_url(chain_spec_url, madara_path.clone() + "/chain-specs")?;
+    if cli.run.run_cmd.shared_params.chain.is_some() {
+        cli.run.run_cmd.rpc_external = true;
+        cli.run.run_cmd.rpc_methods = RpcMethods::Unsafe;
     }
 
+    Ok(())
+}
+
+fn set_chain_spec(cli: &mut Cli, chain_spec_url: String) -> Result<(), String> {
+    let madara_path = cli.run.madara_path.clone().expect("Failed retrieving madara_path").to_str().unwrap().to_string();
+
+    utils::fetch_from_url(chain_spec_url.clone(), madara_path.clone() + "/chain-specs")?;
+    let chain_spec = chain_spec_url.split('/').last().expect("Chain spec file name not found");
+    cli.run.run_cmd.shared_params.chain = Some(madara_path + "/chain-specs/" + chain_spec);
+
+    Ok(())
+}
+
+fn fetch_madara_configs(cli: &Cli) -> Result<(), String> {
+    let madara_path = cli.run.madara_path.clone().expect("Failed retrieving madara_path").to_str().unwrap().to_string();
```

Comment:
This exact code appears three time, maybe it could be encapsulated in its own function
https://en.wikipedia.org/wiki/Rule_of_three_(computer_programming)

Reply:
Is it worth having a function for return the madara path as a string?